### PR TITLE
libqalculate, qalculate-gtk: update to 3.12

### PR DESCRIPTION
**Pull request by judaew**
#### Description

Update to 3.12

###### Tested on

macOS 10.15.6
Xcode 11.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: math/libqalculate/Portfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 PortSystem          1.0
 PortGroup           github 1.0
 
-github.setup        qalculate libqalculate 3.3.0 v
+github.setup        qalculate libqalculate 3.12.0 v
 revision            1
```
Comment (mf2k) [inline math/libqalculate/Portfile:7]: The `revision` needs to be set back to 0 when increasing the `version`.

Comment (mf2k) [reply math/libqalculate/Portfile:7]: Also, the `epoch` may need to be added and set to 1 because "12" is alphabetically less than "3".

**Comment by judaew** [comment]
Thanks, done.

**Comment by mf2k** [comment]
Ideally this would be 2 commits, one for each Portfile.

**Comment by mf2k** [comment]
I'm OK with committing this when the builders finish.

**Comment by judaew** [comment]
> Ideally this would be 2 commits, one for each Portfile.

I'll take this into account. Thanks again for your support.

It just wasn't so important on nixOS.

**Comment by jjstickel** [comment]
LGTM. Thanks.

**Comment by ryandesign** [comment]
MacPorts knows how to do version number comparisons. You only increase the epoch if the version number is changing in a way that MacPorts can't recognize as a normal increase (for example if you were going from a date-based version number like 20200803 to 1.0.0). You can always run the vercmp procedure yourself to see what it says to decide if increasing the epoch is needed, for example using [my wrapper script](https://github.com/ryandesign/macports-user-ryandesign/blob/master/scripts/vercmp).

**Pull request merged**